fagiServer/src/test/java/com/fagi/worker/InputWorkerTests.java

```diff
@@ -0,0 +1,431 @@
+package com.fagi.worker;
+
+import com.fagi.encryption.AES;
+import com.fagi.encryption.AESKey;
+import com.fagi.encryption.Conversion;
+import com.fagi.encryption.Encryption;
+import com.fagi.encryption.RSA;
+import com.fagi.handler.ConversationHandler;
+import com.fagi.handler.InputHandler;
+import com.fagi.model.Data;
+import com.fagi.model.Login;
+import com.fagi.model.Session;
+import com.fagi.model.UserNameAvailableRequest;
+import com.fagi.responses.AllIsWell;
+import com.fagi.util.OutputAgentTestUtil;
+import com.fagi.worker.running.IsWorkerRunningStrategy;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.ByteArrayOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.PrintStream;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+public class InputWorkerTests {
+    private static final byte[] ENCRYPTED_DATA = "encryptedData".getBytes();
+    private OutputWorker outputWorker;
+    private ConversationHandler conversationHandler;
+    private Data data;
+    private IsWorkerRunningStrategy runOnceStrategy;
+    private InputWorker inputWorker;
+    private ObjectInputStream mockObjectInputStream;
+
+    @BeforeEach
+    void setup() throws IOException, ClassNotFoundException {
+        mockObjectInputStream = Mockito.mock(ObjectInputStream.class);
+        when(mockObjectInputStream.readObject()).thenReturn(ENCRYPTED_DATA);
+
+        runOnceStrategy = new WorkerRunOnceStrategy();
+
+        outputWorker = Mockito.mock(OutputWorker.class);
+        conversationHandler = Mockito.mock(ConversationHandler.class);
+        data = Mockito.mock(Data.class);
+
+        inputWorker = new InputWorker(
+                mockObjectInputStream,
+                outputWorker,
+                conversationHandler,
+                data
+        );
+    }
+
+    @AfterEach
+    void tearDown() {
+        System.setErr(System.err);
```

Comment:
What about a try finally in the specific tests? So each test is responsible for cleaning up.